lib/improver/tests/acceptance/acceptance.py

```diff
@@ -30,17 +30,73 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """Setup and checking of known good output for CLI tests"""
 
+import importlib
 import os
 import pathlib
+import shlex
 import shutil
 import subprocess
 import tempfile
 
 import pytest
 
-TIGHT_TOLERANCE = 1e-5
-DEFAULT_TOLERANCE = 1e-4
-LOOSE_TOLERANCE = 1e-2
+from improver import cli
+
+TIGHT_TOLERANCE = 1e-4
+DEFAULT_TOLERANCE = 1e-2
+LOOSE_TOLERANCE = 1e-1
+
+
+# pylint: disable=too-few-public-methods
+class RunCLI():
+    """
+    Wrapper for run_improver_cli to avoid repetition of the CLI name in each
+    call.
+    Equivalent to functools.partial, but nicer to read and avoids functools
+    import for once off use in all test modules.
+    """
+    def __init__(self, cli_name):
+        self.cli_name = cli_name
+
+    def __call__(self, args):
+        run_improver_cli(self.cli_name, args)
+
+
+def run_improver_cli(cli_name, args):
+    """
+    Run an IMPROVER CLI in an acceptance test.
+    Converts arguments to strings and prints the command line for easy
+    re-running by copy-pasting from test output.
+
+    Args:
+        cli_name (str): name of the command line interface to run
+        args (Iterable[Object]): command line arguments, will be converted
+            to a list of strings before passing to CLI functions
+
+    Returns:
+        None
+    """
+    args_strings = stringify(args, print_copyable=False)
+    cli_args = ["improver", cli_name, *args_strings]
+    print(' '.join([shlex.quote(str(x)) for x in cli_args]))
+    cli.run_main(cli_args)
+
+
+def stringify(itr, print_copyable=True):
+    """
+    Convert iterable to a list of strings.
+
+    Args:
+        itr (Iterable[Object]): iterable of generic objects
+        print_copyable (bool): prints a copy-pastable string into test logs
+
+    Returns:
+        List[str]: list of items as strings
+    """
+    it_str = [str(x) for x in itr]
+    if print_copyable:
+        print(' '.join([shlex.quote(str(x)) for x in it_str]))
+    return it_str
```

Comment:
I think most of this functionality might be already included in `improver.cli` module (printing with `improver --verbose` (might need to put that code path before command execution and add shlex.quote), `stringify` - see `'improver.cli.ObjectAsStr`). Might be better to do any additions there, rather than add more wrappers here.

Reply:
I believe that stingify is only being used with argparser and will be redundant once clize is being used everywhere.

Reply:
Nope. It's used a lot for the old argparser CLIs, but it's also used in `run_improver_cli` which is specifically for the new CLIs.

@tjtg, if you are editing things, I wonder if it's not worth putting `run_improver_cli` inside the `RunCLI` class, as I don't think it's being used elsewhere? (Unless you foresee a situation where we might want to use it without the class)

Reply:
I've changed `run_cli` to be a function. The `stringify` and `run_improver_cli` functions are gone since everything is being called via clize now.